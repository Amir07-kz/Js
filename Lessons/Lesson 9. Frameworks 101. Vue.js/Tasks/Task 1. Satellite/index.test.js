/**
 * @jest-environment jsdom
 */

const { mount } = require("@vue/test-utils");
const Satellite = require("./Satellite.vue");

describe("c.9.1 Satellite", () => {
    const wrapper = mount(Satellite.default);
    expect(wrapper.element).toMatchSnapshot();

    it("c.9.1.1 Satellite has data", () => {
        expect(typeof Satellite.default.data).toBe("function");
    });

    it("c.9.1.2 Satellite is a Vue instance", () => {
        expect(wrapper.vm).toBeTruthy();
    });

    const input = wrapper.find('.form-control');

    it("c.9.1.3 with radius 42000k", async () => {
        input.element.value = 42000;
        await input.trigger("input");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("c.9.1.4 with radius 101010k", async () => {
        input.element.value = 101010;
        await input.trigger("input");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("c.9.1.5 with radius 7400k", async () => {
        input.element.value = 7400;
        await input.trigger("input");
        expect(wrapper.element).toMatchSnapshot();
    });
});
