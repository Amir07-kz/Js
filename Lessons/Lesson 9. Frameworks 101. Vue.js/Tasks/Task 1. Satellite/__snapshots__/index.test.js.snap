// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="satellite"
>
  <div
    class="satellite__radius"
  >
    <span>
      Радиус орбиты спутника:
    </span>
     
    <input
      class="form-control"
      type="text"
    />
     
    <span>
      км
    </span>
  </div>
   
  <div
    class="satellite__speed"
  >
    <h3>
      Требуемая скорость спутника:
    </h3>
     
    <div>
      
            Infinity м/c
        
    </div>
     
    <img
      src="https://mediasat.info/wp-content/uploads/2015/06/sat-orbit.jpg"
    />
  </div>
</div>
`;

exports[`c.9.1 Satellite c.9.1.3 with radius 42000k 1`] = `
<div
  class="satellite"
>
  <div
    class="satellite__radius"
  >
    <span>
      Радиус орбиты спутника:
    </span>
     
    <input
      class="form-control"
      type="text"
    />
     
    <span>
      км
    </span>
  </div>
   
  <div
    class="satellite__speed"
  >
    <h3>
      Требуемая скорость спутника:
    </h3>
     
    <div>
      
            3087 м/c
        
    </div>
     
    <img
      src="https://mediasat.info/wp-content/uploads/2015/06/sat-orbit.jpg"
    />
  </div>
</div>
`;

exports[`c.9.1 Satellite c.9.1.4 with radius 101010k 1`] = `
<div
  class="satellite"
>
  <div
    class="satellite__radius"
  >
    <span>
      Радиус орбиты спутника:
    </span>
     
    <input
      class="form-control"
      type="text"
    />
     
    <span>
      км
    </span>
  </div>
   
  <div
    class="satellite__speed"
  >
    <h3>
      Требуемая скорость спутника:
    </h3>
     
    <div>
      
            1990 м/c
        
    </div>
     
    <img
      src="https://mediasat.info/wp-content/uploads/2015/06/sat-orbit.jpg"
    />
  </div>
</div>
`;

exports[`c.9.1 Satellite c.9.1.5 with radius 7400k 1`] = `
<div
  class="satellite"
>
  <div
    class="satellite__radius"
  >
    <span>
      Радиус орбиты спутника:
    </span>
     
    <input
      class="form-control"
      type="text"
    />
     
    <span>
      км
    </span>
  </div>
   
  <div
    class="satellite__speed"
  >
    <h3>
      Требуемая скорость спутника:
    </h3>
     
    <div>
      
            7354 м/c
        
    </div>
     
    <img
      src="https://mediasat.info/wp-content/uploads/2015/06/sat-orbit.jpg"
    />
  </div>
</div>
`;
